### PR TITLE
DAH-1151 Add image card subtitle

### PR DESCRIPTION
**Pull request by christine-sfg**
DAH-1151

Add/style following fields under the image card for mobile and desktop and all languages
-Listing name
-Address
-Developer
-View on Map link
-Share link

image
image

### CODE REVIEW

**Review by kramduckner — APPROVED** [review]

File: app/javascript/components/ListingAddress.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+export const ListingAddress = ({ listing }: ListingAddressProps) => {
+  return (
+    listing.Building_Street_Address &&
+    listing.Building_City &&
+    listing.Building_State &&
+    listing.Building_Zip_Code && (
+      <span>
+        {listing.Building_Street_Address}, {listing.Building_City},{" "}
+        <abbr title="California">CA</abbr> {listing.Building_Zip_Code}
```
Comment (jacksongonzales) [inline app/javascript/components/ListingAddress.tsx:16]: if we wanna nix the annoying text decoration here, I think you can add `className="no-underline"` to the `abbr`. and sorry for not adding on my PR 😭

File: app/javascript/modules/listingDetails/ListingDetailsImageCard.tsx [hunk 2]
```diff
@@ -19,16 +28,35 @@ export const ListingDetailsImageCard = ({ listing }: ListingDetailsImageCardProp
...
+                {t("label.viewOnMap")}
+              </a>
+            </p>
+          </div>
+          <div>
+            <a href={`${getShareListingPath()}/${listing.listingID}`} target="_blank">
+              <img
+                alt={t("label.share")}
+                onMouseEnter={(e) => (e.currentTarget.src = shareButtonSelected)}
+                onMouseLeave={(e) => (e.currentTarget.src = shareButton)}
```
Comment (jacksongonzales) [inline app/javascript/modules/listingDetails/ListingDetailsImageCard.tsx:54]: we should be able to accomplish this behavior (I think, not _positive_ an `img` tag won't act funky) while also getting similar interaction w/ keyboard users by doing 

```suggestion
                onFocus={(e) => (e.currentTarget.src = shareButtonSelected)}
                onBlur={(e) => (e.currentTarget.src = shareButton)}
```
you might also need to add `tabindex="0"` tho it already appears to be tab-able, so maybe unnecessary

File: app/javascript/util/listingUtil.ts [hunk 3]
```diff
@@ -53,3 +54,20 @@ export const isSale = (listing: RailsRentalListing | RailsSaleListing) =>
...
+ * @returns {string} the full address string if all required fields are present, empty
+ * string otherwise
+ */
+export const getListingAddressString = (listing: RailsListing): string => {
+  return (
+    (listing.Building_Street_Address &&
+      listing.Building_City &&
+      listing.Building_State &&
+      listing.Building_Zip_Code &&
+      `${listing.Building_Street_Address}, ${listing.Building_City}, ${listing.Building_State} ${listing.Building_Zip_Code}`) ||
```
Comment (jacksongonzales) [inline app/javascript/util/listingUtil.ts:70]: does this also need the `abbr` tag around the state?

**Review by jacksongonzales — CHANGES_REQUESTED** [review]

**Review by jacksongonzales — APPROVED** [review]
🎉

File: app/javascript/modules/listingDetails/ListingDetailsImageCard.tsx (hunk 2 above)
Comment (christine-sfg) [reply app/javascript/modules/listingDetails/ListingDetailsImageCard.tsx]: @jacksongonzales i updated the event handling so the image should change on hover and also when tabbing to the link

**Comment by christine-sfg** [comment]
Legacy tests are not playing nice this afternoon. Tried re-running several times with no luck. There are no changes here related to the application flow legacy tests so I'm going to go ahead and merge.

**Pull request merged**